pkg/storage/replica_test.go

```diff
@@ -918,16 +916,15 @@ func TestReplicaTSCacheLowWaterOnLease(t *testing.T) {
 		// The other store tries again, this time without the overlap.
 		{storeID: tc.store.StoreID() + 1,
 			start: now.Add(31, 0), expiration: now.Add(50, 0),
-			expLowWater: newLowWater},
-		// Lease is regranted to this replica. Store clock moves forward avoid
-		// influencing the result.
+			// The cache now moves to this other store, and we can't query that.
+			expLowWater: 0},
+		// Lease is regranted to this replica. The low-water mark is updated to the
+		// beginning of the lease.
 		{storeID: tc.store.StoreID(),
 			start: now.Add(60, 0), expiration: now.Add(70, 0),
-			expLowWater: newLowWater},
-		// Lease is held by another once more.
-		{storeID: tc.store.StoreID() + 1,
-			start: now.Add(70, 0), expiration: now.Add(90, 0),
-			expLowWater: newLowWater},
```

Comment:
well it's not testing anything new versus `The other store tries again, this time without the overlap`. Also this test is all about the state of the cache on the first replica; tests that move it away are only useful if the lease is moved back.